polkadot/node/core/pvf/src/prepare/worker_interface.rs

```diff
@@ -232,8 +229,14 @@ async fn handle_response(
 		return Outcome::TimedOut
 	}
 
-	let artifact_id = ArtifactId::from_pvf_prep_data(pvf);
-	let artifact_path = artifact_id.path(cache_path, &checksum);
+	let unique_id = {
+		use rand::RngCore;
+		let mut bytes = [0u8; 64];
+		rand::thread_rng().fill_bytes(&mut bytes);
+		hex::encode(&bytes)
+	};
+	let file_name = format!("{}_0x{}", unique_id, checksum);
+	let artifact_path = cache_path.join(file_name);
```

Comment:
This is now the only place that the path is created. It gets returned to the host and stored in the `Artifacts` table.

I kept the checksum as it will still be needed by execution for https://github.com/paritytech/polkadot-sdk/issues/2399 - but now that I look at it, probably better to instead add the checksum to `ArtifactPathId`.